go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			flag, err := LinkRouteExists(tc.inputLink, tc.inputGwIP, tc.inputSubnet)
+			t.Log(flag, err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
+			if tc.outBoolFlag {
+				assert.True(t, flag)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLinkNeighAdd(t *testing.T) {
+	mockNetLinkLibOperations := new(mocks.NetLinkLibOperations)
+	mockLink := new(netlink_mocks.Link)
+	// below is defined in net_linux.go
+	netLinkOperations = mockNetLinkLibOperations
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputNeigIP              net.IP
+		inputMacAddr             net.HardwareAddr
+		errExp                   bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+		onRetArgsLinkIfaceOpers  []onCallReturnArgs
+	}{
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "test code path with link argument set to nil",
+			errExp: true,
+		},*/
+		// NOTE: since, we dont validate function arguments in the funciton body, a nil value passed for neighIP and neighMac is sufficient
+		{
+			desc:      "test code path where adding neighbor returns an error",
+			inputLink: mockLink,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"NeighAdd", []string{"*netlink.Neigh"}, []interface{}{fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName", Index: 1}}},
+			},
+		},
+		{
+			desc:      "test code path where adding neighbor returns success",
+			inputLink: mockLink,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"NeighAdd", []string{"*netlink.Neigh"}, []interface{}{nil}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName", Index: 1}}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			err := LinkNeighAdd(tc.inputLink, tc.inputNeigIP, tc.inputMacAddr)
+			t.Log(err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
```

Comment:
missing the `else` part and in other test cases as well? please check.

Reply:
shall confirm and fix accordingly